### PR TITLE
Rachel - L141.hfc_R_S_T_Y

### PR DESCRIPTION
**Pull request by rhoesly**
#423

### CODE REVIEW

**Comment by bpbond** [comment]
Note oldnew test is not passing.

**Comment by bpbond** [comment]
Please silence package check notes:

```
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘year’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘value’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘GCAM_region_ID’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘sector’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘fuel’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘service’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘IPCC_description’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘agg_sector’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘ISO_A3’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘iso’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘EDGAR_agg_sector’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘Non.CO2’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘emissions’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘Sector’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘.’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘total’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘share’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘Species’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘Emissions’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘Guus_tot’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘EDGAR_tot’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘scaler’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘supplysector’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘subsector’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘stub.technology’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘adj_emissions’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘energy’
module_emissions_L141.hfc_R_S_T_Y: no visible binding for global
  variable ‘em_fact’
```

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 1]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      rename(EDGAR_agg_sector=Sector)
+    L141.EDGAR_hfc_R_S_T_Yh.long <- bind_rows( L141.EDGAR_hfc_R_S_T_Yh_rest,
+                                               L141.EDGAR_hfc_R_S_T_Yh_other ) %>%
+      mutate(year = as.numeric(year))
+
+    # Map Emissions to GCAM technologies
+    L141.hfc_R_S_T_Yh.long <- gcam_fgas_tech %>%
+      repeat_add_columns(tibble::tibble(GCAM_region_ID = unique(GCAM_region_names$GCAM_region_ID))) %>%
+      repeat_add_columns(tibble::tibble(year = emissions.EDGAR_YEARS)) %>%
+      repeat_add_columns(tibble::tibble(Non.CO2 = unique(L141.EDGAR_hfc_R_S_T_Yh.long$Non.CO2))) %>%
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: Does `repeat_add_column(tibble(GCAM_region_ID = ..., year = ..., Non.CO2 = ...))` not work?

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 2]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      left_join(L141.EDGAR_hfc_R_S_T_Yh.long,
+                by = c("EDGAR_agg_sector", "GCAM_region_ID", "year", "Non.CO2")) %>% # there should be NAs in the match.Some emissions sectors have zero F gass emissions
+      mutate_at(vars(),funs(replace(.,is.na(.),0)))
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: This is really unclear--or rather, really inconsistent with rest of data system. Can you use `replace_na` at least?

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 3]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+                                               L141.EDGAR_hfc_R_S_T_Yh_other ) %>%
+      mutate(year = as.numeric(year))
+
+    # Map Emissions to GCAM technologies
+    L141.hfc_R_S_T_Yh.long <- gcam_fgas_tech %>%
+      repeat_add_columns(tibble::tibble(GCAM_region_ID = unique(GCAM_region_names$GCAM_region_ID))) %>%
+      repeat_add_columns(tibble::tibble(year = emissions.EDGAR_YEARS)) %>%
+      repeat_add_columns(tibble::tibble(Non.CO2 = unique(L141.EDGAR_hfc_R_S_T_Yh.long$Non.CO2))) %>%
+      left_join(L141.EDGAR_hfc_R_S_T_Yh.long,
+                by = c("EDGAR_agg_sector", "GCAM_region_ID", "year", "Non.CO2")) %>% # there should be NAs in the match.Some emissions sectors have zero F gass emissions
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: "there should be NAs in the match, as some emissions sectors have zero F gas emissions"

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 4]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      mutate_at(vars(),funs(replace(.,is.na(.),0)))
+
+    # Disaggregate cooling emissions to residential and commercial sectors
+    # Calculate share of res/com cooling emissions from L144.in_EJ_R_bld_serv_F_Yh
+    # Apply share to HFC emissions
+
+    # Select residential and cooling emisssions from L144.in_EJ_R_bld_serv_F_Yh
+    L141.R_cooling_T_Yh.long <- L144.in_EJ_R_bld_serv_F_Yh %>%
+      filter(service %in% c("comm cooling", "resid cooling") & fuel == "electricity")
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: Comma not `&`

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 5]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+    # Group by GCAM region and ID and year in new data frame (use to calculate share of total later)
+    L141.R_cooling_Yh <-  L141.R_cooling_T_Yh.long %>%  group_by(GCAM_region_ID,year) %>%
+      summarize(total = sum(value))
+    # join in the totals we just calculated and calulate the shares
+    L141.R_cooling_T_Yh.long <- left_join_error_no_match(L141.R_cooling_T_Yh.long,L141.R_cooling_Yh, by = c("GCAM_region_ID", "year")) %>%
+      mutate(share = value / total ) %>%
+      mutate(year = as.numeric(gsub('X','',year))) %>%
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: Combine into single mutate. Why are there Xyears?!? Shouldn't be any.

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 6]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      select(GCAM_region_ID,year,service,share,value)
+
+    # Add res/com cooling share to HFC and make adjustment
+    L141.hfc_R_S_T_Yh_share<- left_join(L141.hfc_R_S_T_Yh.long,
+                                        L141.R_cooling_T_Yh.long ,
+                                        by = c("GCAM_region_ID", "year", "supplysector" = "service")) %>% # there should be NAs.
+                                        #there are sectors that don't have emissions
+      mutate_at(vars(),funs(replace(.,is.na(.),1))) %>% # replace those shares with "1"
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: See above

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 7]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      mutate(emissions = emissions*share)
+
+    # add EDGAR 2008 year, as 2010 to match with Guus
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: Guus?

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 8]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+    TEMP <- subset( L141.hfc_R_S_T_Yh_share, year == 2008 )
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: Again, and below, no GCAM-style spacing

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 9]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+    TEMP$year <- 2010
+    L141.hfc_R_S_T_Yh_share <- bind_rows( L141.hfc_R_S_T_Yh_share, TEMP )
+
+    # process Guus inventory
+    HFC_Inventory_GV <- HFC_Inventory_GV %>% mutate(Species = gsub("-","", Species)) %>%
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: Space after comma

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 10]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      mutate(Species = gsub("4310mee", "43",Species)) %>%
+      dplyr::rename(Guus_tot = Emissions)
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: `dplyr::` not needed

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 11]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+
+    # add Guus values to HFC emissions then calculate scalar
+    L141.hfc_scaler <- L141.hfc_R_S_T_Yh_share %>% group_by(year,Non.CO2) %>%
+      summarize(EDGAR_tot = sum(emissions))%>%
+      arrange(Non.CO2,year) %>%
+      left_join(HFC_Inventory_GV, by = c(year = "Year", Non.CO2 = "Species")) %>% # some entries not in GUUS data, default to scaler =1
+      mutate(scaler = Guus_tot/EDGAR_tot) %>%
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: Spaces around math operators please

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 12]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      mutate_at(vars(), funs(replace(., is.na(.), 1))) %>%
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: See above

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 13]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      mutate_at(vars(), funs(replace(., is.infinite(.), 0))) %>%
+      select(year, Non.CO2, scaler)
+
+    # Add scalar to HFC emissions, calculate adjusted emissions
+    L141.hfc_R_S_T_Yh <- L141.hfc_R_S_T_Yh_share %>% left_join(L141.hfc_scaler,
+                                        by = c("year", "Non.CO2") ) %>% # there are NA's, not all sectors have emissions/scalers
+      mutate(adj_emissions = emissions*scaler) %>%
+      group_by(GCAM_region_ID, supplysector, subsector, stub.technology, Non.CO2, year) %>% #this takes SOOOO LONG
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: Interesting! Really? Let's take a look in the profiler and see what's going on.

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 14]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      left_join_error_no_match(L141.R_cooling_T_Yh.long %>% select(GCAM_region_ID, year,service,value),
+                               by = c("GCAM_region_ID", "year","supplysector"="service")) %>%
+      rename(energy = value) %>%
+      mutate(em_fact = adj_emissions / energy) %>%
+      select(GCAM_region_ID, supplysector, subsector, stub.technology, Non.CO2, year, em_fact) %>%
+      mutate_at(vars(), funs(replace(., is.na(.), 0))) %>%
+      rename(value = em_fact)
+
+    # final processing - rename a "value" column
+    L141.hfc_R_S_T_Yh <- L141.hfc_R_S_T_Yh %>%
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: Why not put this up above?

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 15]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+   L141.hfc_R_S_T_Yh %>%
+      add_title("HFC emissions by region / sector / technology / gas / historical year") %>%
+      add_units("Unit = Gg") %>%
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: Just "Gg" (applies below too)

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 16]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+                     "emissions/EDGAR/EDGAR_HFC152a",
+                     "emissions/EDGAR/EDGAR_HFC227ea",
+                     "emissions/EDGAR/EDGAR_HFC23",
+                     "emissions/EDGAR/EDGAR_HFC236fa",
+                     "emissions/EDGAR/EDGAR_HFC245fa",
+                     "emissions/EDGAR/EDGAR_HFC32",
+                     "emissions/EDGAR/EDGAR_HFC365mfc",
+                     "emissions/EDGAR/EDGAR_HFC43",
+                     "emissions/HFC_Inventory_GV") %>%
       # typical flags, but there are others--see `constants.R`
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: Remove

File: inst/extdata/emissions/HFC_Inventory_GV.csv [hunk 17]
```diff
@@ -0,0 +1,351 @@
+# File: HFC_Inventory_GV.csv
+# Title: F gases historical trend (SSP2_GV)
+# Units: kt
+# Source:
+# Description:
```
Comment (bpbond) [inline inst/extdata/emissions/HFC_Inventory_GV.csv]: Why are these two lines blank? Source, at least, needs to be specified.

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 18]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+    EDGAR_HFC236fa$Non.CO2 <- "HFC236fa"
+    EDGAR_HFC245fa$Non.CO2 <- "HFC245fa"
+    EDGAR_HFC32$Non.CO2 <- "HFC32"
+    EDGAR_HFC365mfc$Non.CO2 <- "HFC365mfc"
+    EDGAR_HFC43$Non.CO2 <- "HFC43"
+
+    # Convert year columns to numeric values, then combine into one tibble
+    # define function to convert data to numeric values
+    years_to_numeric <- function(xx){
+      xx[as.character(1970:2000)] <- sapply(xx[as.character(1970:2000)],as.numeric)
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: See [comment below](https://github.com/JGCRI/gcamdata/pull/627#discussion_r132336327).

File: R/zchunk_L141.hfc_R_S_T_Y.R (hunk 13 above)
Comment (rplzzz) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: My guess would be failure to ungroup before running the `mutate_at` on line 182.  Ungrouping and using `replace_na` should help.

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 19]
```diff
@@ -1,42 +1,39 @@
...
+#' adjusted to match Guus Velders HFC inventory
 #'
 #' @param command API command to execute
 #' @param ... other optional parameters, depending on command
 #' @return Depends on \code{command}: either a vector of required inputs,
-#' a vector of output names, or (if \code{command} is "MAKE") all
+#' a vectdriveor of output names, or (if \code{command} is "MAKE") all
 #' the generated outputs: \code{L141.hfc_R_S_T_Yh}, \code{L141.hfc_ef_R_cooling_Yh}. The corresponding file in the
 #' original data system was \code{L141.hfc_R_S_T_Y.R} (emissions level1).
 #' @details Describe in detail what this chunk does.
```
Comment (cahartin) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: details please.

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 20]
```diff
@@ -1,42 +1,39 @@
 #' module_emissions_L141.hfc_R_S_T_Y
 #'
-#' Briefly describe what this chunk does.
+#' Calculated HFC emissions from EDGAR, by residential and commercial cooling shares,
+#' adjusted to match Guus Velders HFC inventory
```
Comment (cahartin) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: who is Guus Velders? a paper?

**Comment by rhoesly** [comment]
Yea, when I tried to merge everything went crazy. I was definitely passing tests before. I'll just open a fresh branch, recommit, and open new PR for that branch after I address the comments on the code.

File: R/zchunk_L141.hfc_R_S_T_Y.R (hunk 20 above)
Comment (rhoesly) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: That's what it seems like. It's the "citation" on another inventory we use, but I can't find the actual citation. Was going to open an Issue.

File: R/zchunk_L141.hfc_R_S_T_Y.R (hunk 18 above)
Comment (cahartin) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: can you do something like: `as.numeric(year)` in the pipeline?

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 21]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+                    EDGAR_HFC152a, EDGAR_HFC227ea, EDGAR_HFC23,
+                    EDGAR_HFC236fa, EDGAR_HFC245fa, EDGAR_HFC32,
+                    EDGAR_HFC365mfc, EDGAR_HFC43)
+
+    # Use function to convert datat to numeric values and combine to one tibble
+    # Map to GCAM countries and technologies
+    F_gases_numeric <- lapply(F_gases, years_to_numeric)
+    L141.EDGAR_HFC <- do.call(bind_rows,F_gases_numeric) %>%
+      left_join_error_no_match(EDGAR_sector %>% select(-IPCC_description), by = "IPCC") %>% # Add Edgar agg_sector
+      mutate(EDGAR_agg_sector = agg_sector) %>% #rename agg_sector to EDGAR_agg_sector
```
Comment (cahartin) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: rename is clearer than mutate when you are only renaming the column

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 22]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      left_join(HFC_Inventory_GV, by = c(year = "Year", Non.CO2 = "Species")) %>% # some entries not in GUUS data, default to scaler =1
+      mutate(scaler = Guus_tot/EDGAR_tot) %>%
+      mutate_at(vars(), funs(replace(., is.na(.), 1))) %>%
+      mutate_at(vars(), funs(replace(., is.infinite(.), 0))) %>%
+      select(year, Non.CO2, scaler)
+
+    # Add scalar to HFC emissions, calculate adjusted emissions
+    L141.hfc_R_S_T_Yh <- L141.hfc_R_S_T_Yh_share %>% left_join(L141.hfc_scaler,
+                                        by = c("year", "Non.CO2") ) %>% # there are NA's, not all sectors have emissions/scalers
+      mutate(adj_emissions = emissions*scaler) %>%
```
Comment (cahartin) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: space around `*`

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 23]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      group_by(GCAM_region_ID, supplysector, subsector, stub.technology, Non.CO2, year) %>% #this takes SOOOO LONG
+      summarise(adj_emissions = sum(adj_emissions, na.rm = T)) %>%
+      mutate_at(vars(), funs(replace(., is.na(.), 0))) #replace NAs with zero. Some sectors have zero emissions.
```
Comment (cahartin) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: `replace_na`

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 24]
```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+
+    #Compute final cooling HFC emissions factors
+    L141.hfc_ef_R_cooling_Yh <- L141.hfc_R_S_T_Yh %>%
+      subset( supplysector %in% c( "comm cooling", "resid cooling" ) ) %>%
+      left_join_error_no_match(L141.R_cooling_T_Yh.long %>% select(GCAM_region_ID, year,service,value),
+                               by = c("GCAM_region_ID", "year","supplysector"="service")) %>%
+      rename(energy = value) %>%
+      mutate(em_fact = adj_emissions / energy) %>%
+      select(GCAM_region_ID, supplysector, subsector, stub.technology, Non.CO2, year, em_fact) %>%
+      mutate_at(vars(), funs(replace(., is.na(.), 0))) %>%
```
Comment (cahartin) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: `replace_na`

File: R/zchunk_L141.hfc_R_S_T_Y.R (hunk 20 above)
Comment (cahartin) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: sounds good. open an issue on that one.

File: R/zchunk_L141.hfc_R_S_T_Y.R (hunk 1 above)
Comment (rhoesly) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: Nope... I've tried different variations of the location of the tibble functions and I get different errors. Variable lengths are wrong, unused arguments...etc.

`Browse[2]> L141.hfc_R_S_T_Yh.long <- gcam_fgas_tech %>%
+     repeat_add_columns(tibble::tibble(GCAM_region_ID = unique(GCAM_region_names$GCAM_region_ID),
+                                       year = emissions.EDGAR_YEARS,
+                                       Non.CO2 = unique(L141.EDGAR_hfc_R_S_T_Yh.long$Non.CO2)))
Error: Variables must be length 1 or 38.
Problem variables: 'GCAM_region_ID', 'Non.CO2'`

Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: Huh, ok!

**Comment by rhoesly** [comment]
All comments here are *hopefully* addressed in a new PR from a clean branch (rachel - L141.hfc R S T Y - take2) which has no conflicts.

**Comment by bpbond** [comment]
Thanks @rhoesly . Closing this given that you've opened #631 .

**Pull request closed without merging**